packages/blocks/src/store/utils.js

```diff
@@ -18,3 +18,15 @@ export const getValueFromObjectPath = ( object, path, defaultValue ) => {
 	} );
 	return value ?? defaultValue;
 };
+
+export function matchesAttributes( blockAttributes, variation ) {
+	return Object.entries( variation ).every( ( [ key, value ] ) => {
```

Comment:
I'm wondering if we have a util for this already with all the removal of lodash work.. --cc @tyxla 

Maybe `isShallowEqual`? Additionally it seem we would have the same issue with `arrays`.

Reply:
I actually asked @tyxla about this yesterday 😄 Turns out we used to have a util! I added a note about it to the PR description:

> Note that this PR re-introduces the `matchesAttributes` util function (albeit in a different place) that was previously removed by https://github.com/WordPress/gutenberg/pull/58194.

Reply:
(`isShallowEqual` isn't quite the right fit: We only need the given block to match all the variation object attribute's properties, but it's okay if the block's object attribute has additional properties. E.g. in the example from the PR description, if the given block's `metadata.bindings.content` object has an additional `label` property, the variation should still match it.)

Reply:
> Additionally it seem we would have the same issue with `arrays`.

That's a fair point, but I'd rather tackle that separately. It's a bit less clear to me what the desired behavior should be for them, so I think it deserves a bit of dedicated discussion. If a variation has

```json
"attributes": {
    "arrayAttr": [ "a", "b", "c" ]
},
"isActive": [ "arrayAttr" ]
```

then which of the following blocks should we match?

1. `[ "a", "b", "c", "d" ] // Variation contains a subset`
2. `[ "b", "c", "a" ] // Variation matches array items but not array order`

Reply:
Just to close the loop, based on the need @ockham described, I agree that it's best if we reintroduce that util.